courseraresearchexports/containers/client.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Coursera
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Coursera's tools for managing docker containers configured with a
+postgres database.
+"""
+
+from courseraresearchexports.containers import utils
+from courseraresearchexports import exports
+import logging
+import shutil
+import os
+import time
+
+
+COURSERA_DOCKER_LABEL = 'courseraResearchExport'
+COURSERA_LOCAL_FOLDER = os.path.expanduser('~/.coursera/exports/')
+POSTGRES_DOCKER_IMAGE = 'postgres:9.5'
+POSTGRES_INIT_MSG = 'PostgreSQL init process complete; ready for start up.'
+POSTGRES_READY_MSG = 'database system is ready to accept connections'
+
+
+def get_next_available_port(docker_client):
+    """
+    Find next available port to map postgres port to host
+    :param docker_client:
+    :return:
+    """
+    containers = list_all(docker_client)
+    ports = [utils.get_container_host_port(container, docker_client)
+             for container in containers]
+
+    return max(ports) + 1 if ports else 5433
+
+
+def list_all(docker_client):
+    """
+    Return all containers that have Coursera label
+    :param docker_client:
+    """
+    return docker_client.containers(
+        all=True, filters={'label': COURSERA_DOCKER_LABEL})
+
+
+def start(container, docker_client):
+    """
+    Start a docker container containing a research export database. Waits until
+    """
+    logging.info('Starting containers...')
+    docker_client.start(container)
+    # poll logs to see if database is ready to accept connections
+    while POSTGRES_READY_MSG not in docker_client.logs(container, tail=4):
```

Comment:
do you think it's possible to do some kind of progress bar/loading indicator here?